tests/integrators/advection_01.cc

```diff
@@ -0,0 +1,299 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2012 - 2015 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+
+// Test the functions in integrators/laplace.h
+// Output matrices and assert consistency of residuals
+
+#include "../tests.h"
+#include "../test_grids.h"
+
+#include <deal.II/base/logstream.h>
+#include <deal.II/integrators/advection.h>
+#include <deal.II/integrators/laplace.h>
+
+#include <deal.II/fe/fe_dgq.h>
+#include <deal.II/fe/fe_nedelec.h>
+
+using namespace LocalIntegrators::Advection;
+
+template <int dim>
+void test_cell(const FEValuesBase<dim> &fev)
+{
+  const unsigned int n = fev.dofs_per_cell;
+  unsigned int d=fev.get_fe().n_components();
+  FullMatrix<double> M(n,n);
+  std::vector<std::vector<double> > vel(dim,std::vector<double>(1));
+  vel[0][0]=1.;
+  vel[1][0]=1.;
+  if (dim > 2)
+    vel[2][0]=1.;
+  cell_matrix(M,fev,fev,vel);
+  {
+    LogStream::Prefix pre("cell");
+    M.print(deallog,8);
+  }
+
+  Vector<double> u(n), v(n), w(n);
+  std::vector<std::vector<double > >
+  uval    (d,std::vector<double>(fev.n_quadrature_points));
+
+  std::vector<types::global_dof_index> indices(n);
+  for (unsigned int i=0; i<n; ++i)
+    indices[i] = i;
+
+  {
```

Comment:
I trust you that you have checked the output generated here with some analytical values, it's difficult to check these formulas...